### PR TITLE
Make Debug output for Struct_dm_ioctl more readable

### PR DESCRIPTION
**Pull request by agrover**
When formatting `Debug` output, print strings for Struct_dm_ioctl name and uuid fields, instead of arrays of u8, which are verbose and not that useful.

This brought to light the fact that having these arrays be `i8` is non-optimal, so fix that first.

### CODE REVIEW

**Comment by agrover** [comment]
two follow-on commits to fix some clippy warnings and improve output of the dev field.

**Comment by mulkieran** [comment]
This is related to, but does not fix: #202, i.e., we've just flipped from ugly hard-to-read output which tells the whole story to easy-to-read output which hides some information that we could end up wishing we had. But it may be more helpful at this time.

File: src/dm_ioctl.rs [hunk 1]
```diff
@@ -132,9 +135,24 @@ impl fmt::Debug for Struct_dm_ioctl {
...
             .field("padding", &self.padding)
             .field("dev", &self.dev)
-            .field("name", &self.name.to_vec())
-            .field("uuid", &self.uuid.to_vec())
-            .field("data", &self.data.to_vec())
+            .field(
+                "name",
+                &slice_to_null(&self.name)
+                    .map(|s| String::from_utf8_lossy(s))
+                    .unwrap_or_else(|| Cow::Borrowed("")),
```
Comment (mulkieran) [inline src/dm_ioctl.rs]: You can only have ```None``` here if there is no null-terminator to be found in the vec. Probably better to have some special way of indicating that, as
```<no value found>```, or something.

Comment (agrover) [reply src/dm_ioctl.rs]: hmm. This should never happen, name and uuid are supposed to be null-terminated. Let's try `expect()`.

**Comment by agrover** [comment]
* switched to using `expect()`
* Add a commit that omits unused/padding fields from Debug output

**Comment by mulkieran** [comment]
If the only choice is an empty string as a defaykt or an ```expect()```, empty string is the right choice. Perhaps we know that dm always zeroes out all fields that it doesn't fill w/ an explicit value. And we also assume that dm code always puts a null-terminator on the end of the string that it writes in those fields. But, it doesn't make sense to crash just because some dm code has a bug, which is the effect of having an ```expect()``` instead of a default value. After all, we probably are using the Debug implementation because something went wrong. Hopefully it wasn't a badly written UUID on some unlikely code path in the kernel, but we can't be certain of that.

**Comment by mulkieran** [comment]
Please change title to "Make Debug output for Struct_dm_ioctl more readable". "Improve" is a bit too broad, because in some debugging contexts, it will make the output useless. We just hope those debugging contexts are the less usual ones.

**Comment by agrover** [comment]
Thanks! Merging.

**Pull request merged**